ext/TensorKitChainRulesCoreExt.jl

```diff
@@ -172,7 +172,9 @@ end
 function ChainRulesCore.rrule(::typeof(norm), a::AbstractTensorMap, p::Real=2)
     p == 2 || error("currently only implemented for p = 2")
     n = norm(a, p)
-    norm_pullback(Δn) = NoTangent(), a * (Δn' + Δn) / (n * 2), NoTangent()
+    function norm_pullback(Δn)
+        return NoTangent(), a * (Δn' + Δn) / (n * 2 + eps(real(eltype(a)))), NoTangent()
```

Comment:
Could you change this to `a * (Δn' + Δn) / 2 / hypot(n, eps(one(n)))` ? I think that is slightly nicer, in that, if `n==1.`, then `n+eps()` is no longer exactly one, but `hypot(1.,eps())` is still exactly `1.` due to machine precision.

Reply:
Yes, this is indeed better. I have modified this line according to the suggestion.